### DISCUSSION TITLE
GraalVM native image support for Groovy 3?

### DISCUSSION THREAD

**Discussion by dmurat** [micronaut-projects/micronaut-core#4239] [Q&A] [answered]
After looking at the Szymon Stepniak's video about compiling Groovy into the native executable (https://www.youtube.com/watch?v=RPdugI8eZgo), I wonder if Micronaut has plans to add support for Groovy with GraalVM?

If relevant, I'm not a Micronaut user at the moment, but I'm planning to use it with the Groovy 3.x soon.

Tnx.

**Comment by graemerocher** [comment, marked as answer]
Whilst it is possible to get simple things working on native image with Groovy due to the dynamic nature of Groovy and extensive use of reflection it is much harder to achieve. Currently native image is better suited to statically typed languages.

**Reply by dmurat** [reply]
Is this true even if Groovy static compilation is used exclusively?

**Reply by graemerocher** [reply]
even when static compilation is used there are parts of Groovy that still require dynamic dispatch (for example closures)
